spring-integration-core/src/main/java/org/springframework/integration/endpoint/AbstractPollingEndpoint.java

```diff
@@ -343,13 +345,16 @@ public void run() {
 						count++;
 					}
 					catch (Exception e) {
-						if (e instanceof RuntimeException) {
-							throw (RuntimeException) e;
+						if (e instanceof MessagingException) {
+							throw (MessagingException) e;
 						}
 						else {
-							throw new MessageHandlingException(new ErrorMessage(e), e);
+							throw new MessagingException(messageHolder.get(), e);
```

Comment:
Actually, I don't think `MessageHandlingException` is correct here; if the exception occurred on a downstream handler, it will already be an `MHE`; it not, then it's not really an MHE it's an "unknown" type of ME.

I wonder if we should take the opportunity we discussed last week - i.e. an enhanced messaging exception where the payload contains the failedMessage as well as this (`inputMessage`).